src/EnergyPlus/api/runtime.py

```diff
@@ -551,8 +551,24 @@ def callback_after_component_get_input(self, state: c_void_p, f: FunctionType) -
         all_callbacks.append(cb_ptr)
         self.api.callbackEndOfAfterComponentGetInput(state, cb_ptr)
 
-    # user defined component callbacks are not allowed, they are coupled directly to a specific EMS manager/PythonPlugin
-    # def callback_user_defined_component_model(self, f: FunctionType) -> None:
+    def callback_user_defined_component_model(self, state: c_void_p, f: FunctionType, program_name: str) -> None:
```

Comment:
Provide a Python API function for registering a callback.  This is what most people will be using.